internal/domain/entities/metrics.go

```diff
@@ -1,15 +1,12 @@
 package entities
 
-type MetricName string
+import "errors"
 
-type MetricsCollection struct {
-	GaugeMetrics   map[MetricName]Gauge
-	CounterMetrics map[MetricName]Counter
-}
+var ErrJSONMarshal = errors.New("failed to marshal to JSON")
```

Comment:
It's better to define errors (and other entities too) close to the spot where they are used